### ISSUE TITLE
unify 'options' handling for sender and receiver links

### ISSUE THREAD

**Issue by mbroadst** [noodlefrenzy/node-amqp10#149]
Both [AMQPClient.prototype.createSender](https://github.com/noodlefrenzy/node-amqp10/blob/master/lib/amqp_client.js#L180) and [AMQPClient.prototype.createReceiver](https://github.com/noodlefrenzy/node-amqp10/blob/master/lib/amqp_client.js#L232) take `options` as a second parameter, however, it's a bit confusing to the user what those options actually are. I'm opening this issue so we can discuss how best to move forward with these in v2.0.0.

Currently each link accepts options with the following criteria:

**ReceiverLink** 
  * `filter` if filter exists, and filter is an array with a map as the first entry (<-- incredibly specific), then it is set as the source filter on the receiver link
  * `policy` if policy exists, then it is deepMerged onto the existing policy (made up from the Client's receiverLinkPolicy, as well as options specific to the address passed in). 

**SenderLink**
  * `options` aren't used at all, other than to create a unique identifier for the linkName

So the question is: what _should_ `options` actually be for both link types?

**Comment by noodlefrenzy** [comment]
Good summary. Are we still thinking of having a SimpleClient? If so, we could potentially punt on a top-level filter and just have options actually _be_ `policy` overrides, and those who want to specify filters need to pass in
```
{ 
  options: {
    source: {
      filter: ...
} } }
```

**Comment by mbroadst** [comment]
When using the library the other day it seemed to me that `options` should be `attach frame options` (aka `link creation options`), which I totally understand to be restrictive and perhaps too low-level. On the other hand, this _felt_ a little awkward:

```
client.createReceiver(address, {
  policy: {
    options: {
      receiverSettleMode: c.receiverSettleMode.settleOnDisposition
    },
    creditQuantum: 1
  }
});
```

**Comment by noodlefrenzy** [comment]
That would make it more painful for the client, but much more straightforward about what should be in there, and where.

**Comment by noodlefrenzy** [comment]
The other option would be to only allow top-level policy settings in the policy and not attach-frame options, and name attach-frame options something like `attach`, and still allow filter, so it'd look something like:
```
{
  policy: { creditQuantum: 1},
  attach: { receiverSettleMode: ... },
  filter: ...
}
```

**Comment by mbroadst** [comment]
@noodlefrenzy I think that's a good idea (options === policy override). It seems to me that the SimpleClient is an inevitability at this point, not only as a cool example of how to wrap our lower level api, but also to ease migration for v1.0.x users.

**Comment by mbroadst** [comment]
Upon further reflection, I would be totally OK with including "filter" as a temporary key accepted in what is otherwise considered a "linkPolicy" for `options`. This would get us the best of both worlds in that: `options` can generally be considered an alias for `[sender, receiver]LinkPolicy`, and in the special case of the receiver we "pull out" the `filter` if its there and otherwise merge the policy with the DefaultPolicy's linkPolicy. Thoughts?

**Comment by noodlefrenzy** [comment]
I'm with you re: preferring changing `options` to `(receiver|sender)Policy` vs. extending options with attach/etc. I'm less OK with allowing `filter` at the top level. Although I _would_ be OK with `createReceiver(address, receiverPolicy, filter)` - i.e. adding an extra parameter.

**Comment by mbroadst** [comment]
fixed in #151
